### PR TITLE
Create outline for `focus-within` on RadioSwitch

### PR DESCRIPTION
**Pull request by JuliaKirschenheuter**
### ☑️ Resolves

* Fix https://github.com/nextcloud/server/issues/40706

- [x] Equalize styles for focused state: keyboard as well as mouse click
- [x] Apply styles for unchecked state

### 🖼️ Screenshots

|  Before   | After                                                        |
|-------------|---------------------------------------------------------------|
| image | Screenshot from 2023-10-26 13-28-29 |-------------|---------------------------------------------------------------|
| image | Screenshot from 2023-10-26 13-26-04 |-------------|---------------------------------------------------------------|
| Screenshot from 2023-10-26 13-34-43 | Screenshot from 2023-10-26 13-29-24 |-------------|---------------------------------------------------------------|
|  | |-------------|---------------------------------------------------------------|
| Screenshot from 2023-10-26 13-35-20 | Screenshot from 2023-10-26 13-29-14

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable

### CODE REVIEW

**Comment by susnux** [comment]
Does not work on the button style:

image

**Comment by susnux** [comment]
Maybe something like `outline-offset: -2px;`?

**Comment by JuliaKirschenheuter** [comment]
> Does not work on the button style:
image

Do you mean?:

Peek 2023-10-27 13-00

**Comment by ShGKme** [comment]
> Maybe something like `outline-offset: -2px;`?

I always used `border+padding` to make an outline inside an element, never heard about this property 🙈

File: src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue [hunk 1]
```diff
@@ -620,8 +620,12 @@ export default {
 		margin: 4px $icon-margin;
 	}
 
-	&__input:focus-visible + &__content {
-		outline: 2px solid var(--color-primary-element) !important;
+	&__input:focus-visible + &__content,
+	&__content:active,
+	&__input:focus-within + &__content {
```
Comment (Pytal) [inline src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:625]: Are the `:active` and `:focus-within` pseudo-classes necessary? As now the borders are visible when just clicking with a mouse and stay until clicking elsewhere

Comment (JuliaKirschenheuter) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:625]: in my opinion we need it. without: 
Peek 2023-10-30 18-46

**Review by susnux — APPROVED** [review]
Looks good overall, but I agree with @Pytal (while it is not a blocker for me)

File: src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue (hunk 1 above)
Comment (susnux) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:625]: I agree with @Pytal, I think it is a bit weird to have to focus outline also on click, we do not have it for buttons or other input elements.

Comment (Pytal) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:625]: Was previously discussed with @michaelnissenbaum that only keyboard focus required focus borders but if you think this is better then fine with me :)

**Comment by marcoambrosini** [comment]
@JuliaKirschenheuter focus-within is not needed and causes an unwanted behavior of showing the outline on click. We don't want that. I took the liberty to push a fix. Please review @Pytal @susnux @JuliaKirschenheuter 
Thanks :)

**Review by marcoambrosini — DISMISSED** [review]
blocking this until we reach a proper solution

**Comment by JuliaKirschenheuter** [comment]
> @JuliaKirschenheuter focus-within is not needed and causes an unwanted behavior of showing the outline on click. We don't want that.

I have to clarify that. Probably i had mistakes in thinking.

**Review by susnux — APPROVED** [review]
Checked again, works!

**Review by Pytal — APPROVED** [review]

**Comment by skjnldsv** [comment]
Please set appropriate `fix` or `enhancement` labels next time :+1:

**Pull request merged**